src/hotspot/cpu/x86/vm_version_x86.hpp

```diff
@@ -1044,6 +1044,21 @@ class VM_Version : public Abstract_VM_Version {
   static bool supports_clflushopt() { return ((_features & CPU_FLUSHOPT) != 0); }
   static bool supports_clwb() { return ((_features & CPU_CLWB) != 0); }
 
+  // Old CPUs perform lea on AGU which causes additional latency transfering the
+  // value from/to ALU for other operations
+  static bool supports_fast_2op_lea() {
+    return (is_intel() && supports_avx()) || // Sandy Bridge and above
+           (is_amd()   && supports_avx());   // Jaguar and Bulldozer and above
+  }
+
+  // Pre Icelake Intels have inefficient 3-op lea with 3 latency, this can be
+  // replaced by add-add or lea-add
+  static bool supports_fast_3op_lea() {
+    return supports_fast_2op_lea() &&
+           ((is_intel() && supports_clwb()) || // Icelake and above
+            is_amd());
```

Comment:
Does it mean all AMD cpus would be fast-3op if they are fast-2op?

Reply:
Yes, AMD complex `lea` has 2 cycle latency and can be executed on 2 ports so it is better than 2 `add`s.